crates/vm/levm/src/vm.rs

```diff
@@ -592,37 +597,50 @@ impl VM {
     }
 
     /// Gets the max blob gas cost for a transaction that a user is willing to pay.
-    fn get_max_blob_gas_cost(&self) -> Result<U256, VMError> {
-        let blob_gas_used = U256::from(self.env.tx_blob_hashes.len())
+    fn get_max_blob_gas_price(&self) -> Result<U256, VMError> {
+        let blob_gasses = self.env.tx_blob_hashes.len();
+
+        let blob_gas_used: u64 = blob_gasses
+            .try_into()
+            .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
+
+        let blob_gas_used: u64 = blob_gas_used
             .checked_mul(BLOB_GAS_PER_BLOB)
             .unwrap_or_default();
 
         let max_blob_gas_cost = self
             .env
             .tx_max_fee_per_blob_gas
             .unwrap_or_default()
-            .checked_mul(blob_gas_used)
+            .checked_mul(blob_gas_used.into())
             .ok_or(InternalError::UndefinedState(1))?;
 
         Ok(max_blob_gas_cost)
     }
 
     /// Gets the actual blob gas cost.
-    fn get_blob_gas_cost(&self) -> Result<U256, VMError> {
-        let blob_gas_used = U256::from(self.env.tx_blob_hashes.len())
+    fn get_blob_gas_price(&self) -> Result<U256, VMError> {
+        let blob_gas_used: u64 = self
+            .env
+            .tx_blob_hashes
+            .len()
+            .try_into()
+            .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
+
+        let blob_gas_price: u64 = blob_gas_used
             .checked_mul(BLOB_GAS_PER_BLOB)
             .unwrap_or_default();
 
         let base_fee_per_blob_gas = self.get_base_fee_per_blob_gas()?;
 
-        let blob_fee = blob_gas_used
+        let blob_fee = blob_gas_price
             .checked_mul(base_fee_per_blob_gas)
-            .ok_or(InternalError::UndefinedState(1))?;
+            .ok_or(VMError::Internal(InternalError::ConversionError))?;
```

Comment:
Again, if possible don't change things that are not directly related to the PR's objective, specially if it is a huge refactor PR

Reply:
Changed here: 
https://github.com/lambdaclass/ethrex/pull/1528/commits/17e8c68cc07b836fbdb220b0d25695bee237c1e5